tests/phpunit/tests/test-wpml-cornerstone-translatable-nodes.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Test_WPML_Cornerstone_Translatable_Nodes test file.
+ *
+ * @package wpml-page-builders-cornerstone
+ */
+
+/**
+ * Class Test_WPML_Cornerstone_Translatable_Nodes
+ *
+ * @group page-builders
+ * @group cornerstone
+ * @group wpmltm-3081
+ */
+class Test_WPML_Cornerstone_Translatable_Nodes extends OTGS_TestCase {
+
+	/**
+	 * Test get method.
+	 *
+	 * @dataProvider node_data_provider
+	 *
+	 * @param string $type                 Node type.
+	 * @param string $field                Node field.
+	 * @param string $expected_title       Expected title.
+	 * @param string $expected_editor_type Expected editor type.
+	 */
+	public function test_get( $type, $field, $expected_title, $expected_editor_type ) {
+		$wrap_tag = 'h2';
+
+		\WP_Mock::wpPassthruFunction( '__' );
+
+		$node_id  = mt_rand( 1, 100 );
+		$settings = array(
+			'_type' => $type,
+			$field  => rand_str(),
+		);
+		if ( 'heading' === $type ) {
+			$settings['text_tag'] = $wrap_tag;
+		}
+
+		$subject = new WPML_Cornerstone_Translatable_Nodes();
+		$strings = $subject->get( $node_id, $settings );
+		$this->assertCount( 1, $strings );
+		$string = $strings[0];
+		$this->assertEquals( $settings[ $field ], $string->get_value() );
+		$this->assertEquals( $field . '-' . $settings['_type'] . '-' . $node_id, $string->get_name() );
+		$this->assertEquals( $expected_title, $string->get_title() );
+		$this->assertEquals( $expected_editor_type, $string->get_editor_type() );
+		if ( 'heading' === $type ) {
+			$this->assertEquals( $wrap_tag, $string->get_wrap_tag() );
+		}
+	}
+
+	/**
+	 * Data provider for test_get().
+	 *
+	 * @return array
+	 */
+	public function node_data_provider() {
```

Comment:
@kagg-design we prefix data providers with `dp_` as a convention. In this case, I guess the correct name should be `dp_valid_nodes`-

Also, a good test class would include another test with a different provider which passes invalid nodes.

Reply:
@andreasciamanna Data provider renamed. Test for invalid node added.